DS13/code/modules/borg/items+structures.dm

```diff
@@ -0,0 +1,596 @@
+			L += T
+	target = pick(L)
+	if(!target)
+		to_chat(user, "Unable to set translocation target, could not locate a suitable subspace exit point.")
+		return
+	to_chat(user, "Coordinates locked: [target.x],[target.y]. Unimatrix 325, grid 006")
+	icon_state = "teleporter-on"
+*/
+/obj/structure/borg_teleporter/proc/activate()
+	icon_state = "teleporter-off"
+	target = null
+	var/list/A = list()
+	A = GLOB.borg_collective.poll_drones_for_teleport()
+	var/area/thearea
+	if(A.len)
+		thearea = pick(A)
+	if(!A.len || !thearea || QDELETED(thearea))
+		var/xd = pick(GLOB.teleportlocs)
+		thearea = GLOB.teleportlocs[xd]
+	var/list/L = list()
+	for(var/turf/T in get_area_turfs(thearea.type))
+		if(!is_blocked_turf(T) && !istype(T, /turf/open/openspace))
+			L += T
+	target = pick(L)
+	if(!target) //Aaaand it still couldn't find an exit turf....somehow...
+		visible_message("Unable to set translocation target, could not locate a suitable subspace exit point. Contact an admin immediately.")
+		return
+	GLOB.borg_collective.message_collective("The collective", "Translocation matrix coordinates locked: [target.x],[target.y] in [get_area(target)]. Unimatrix 325, grid 006. Step through the teleporter to beam down.")
+	icon_state = "teleporter-on"
+
+//Borg tool
+
+/obj/item/borg_tool
+	name = "Primary prosthetic interaction node"
+	desc = "This massive prosthetic houses a vicious claw and a full replicator suite for the rapid conversion of an area into God knows what. Click it while holding it to change its mode."
+	icon = 'DS13/icons/weapons/melee.dmi'
+	item_state = "borgtool"
+	icon_state = "borgtool"
+	lefthand_file = 'DS13/icons/mob/inhands/weapons/guns_lefthand.dmi'
+	righthand_file = 'DS13/icons/mob/inhands/weapons/guns_righthand.dmi'
+	resistance_flags = UNACIDABLE
+	var/mode = "assimilate"
+	var/convert_time = 30 //3 seconds
+	item_flags = NODROP
+	force = 18 //hella strong
+	var/dismantling_machine = 0
+	var/blacklistedmachines = list(/obj/machinery/computer/communications, /obj/machinery/computer/card)
+	var/saved_time
+	var/cooldown = 10
+	var/resource_amount = 10 //Starts with a bit so you can build a structure from the get-go
+	var/resource_cost = 10
+	var/busy = FALSE //Are you performing an action already? This avoids infinite turf conversion
+
+/obj/item/borg_tool/attack_self(mob/user)
+	var/list/options = list("build", "combat", "assimilate")
+	for(var/option in options)
+		options[option] = image(icon = 'DS13/icons/actions/actions_borg.dmi', icon_state = "[option]")
+	var/dowhat = show_radial_menu(user,user,options)
+	if(!dowhat)
+		return
+	mode = dowhat
+	to_chat(user,"<font color='green' size='2'><b>[src] is now set to [dowhat] mode.</b></font>")
+	playsound(user.loc, 'sound/items/jaws_pry.ogg', 100,1)
+
+/obj/item/borg_tool/attackby(obj/item/stack/I, mob/user)
+	if(istype(I, /obj/item/stack))
+		resource_amount += I.amount
+		to_chat(user, "We have inserted [I] into our [src], its current resource count is: [resource_amount]")
```

Comment:
This will be abused to get a shit ton of materials by inserting tons and tons of sheets of metal and glass

Reply:
TBH, just assimilating is way more efficient anyway. You get 5 mats per tile and the tiles convert in a ring